sdk/include/ddk/ntstrsafe.h

```diff
@@ -2825,6 +2885,121 @@ RtlStringLengthWorkerW(
     return Status;
 }
 
+
+NTSTRSAFEAPI
+RtlUnicodeStringValidate(PCUNICODE_STRING SourceString)
+{
+    if (SourceString->Length % sizeof(WCHAR) != 0 ||
+        SourceString->MaximumLength % sizeof(WCHAR) != 0 ||
+        SourceString->Length > SourceString->MaximumLength ||
+        SourceString->MaximumLength > NTSTRSAFE_UNICODE_STRING_MAX_CCH * sizeof(WCHAR) ||
+        (SourceString->Buffer == NULL && (SourceString->Length != 0 || SourceString->MaximumLength != 0)))
+    {
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    return STATUS_SUCCESS;
+}
+
+NTSTRSAFEAPI
+RtlpUnicodeStringValidateEx(PCUNICODE_STRING SourceString, STRSAFE_DWORD dwFlags)
```

Comment:
Isn't it ccalled "[RtlUnicodeStringValidateEx](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/ntstrsafe/nf-ntstrsafe-rtlunicodestringvalidateex)" without the 'p'?

Reply:
Thanks for the tip, I didn't realize the ValidateEx existed as a public API.